In [3]:
import pandas as pd

In [4]:
titles = pd.read_csv('titles.csv')
titles.head()

title  year
0    Tasveer Mere Sanam  1996
1              Only You  1994
2  El pueblo del terror  1970
3               Machine  2007
4                  MARy  2008

# How many movies are listed in the titles dataframe?

In [5]:
titles.shape[0]

244914

There are 244914 movies in the dataset

# What are the earliest two films listed in the titles dataframe?¶

In [7]:
first = titles.sort_values(by = 'year')
first.head(2)

title  year
193222          Miss Jerry  1894
51388   The Startled Lover  1898

Miss Jerry and The Startled Lover are the first two films listed


# How many movies have the title "Hamlet"?

In [11]:
titles[titles['title'] == "Hamlet"].count()

title    20
year     20
dtype: int64

20 movies have the word HAMLET in their titles.

# How many movies are titled "North by Northwest"?

In [12]:
titles[titles['title'] == "North by Northwest"].count()

title    1
year     1
dtype: int64

There is only one movie that has the name North by Northwest

# When was the first movie titled "Hamlet" made?

In [14]:
ham=titles[titles['title'] == "Hamlet"].sort_values(by='year')
ham.head(1)

title  year
52521  Hamlet  1910

Hamlet was made in 1910

# List all of the "Treasure Island" movies from earliest to most recent.


In [15]:
ti = titles[titles['title'] == "Treasure Island"].sort_values(by='year').values.tolist()
print('List all of the Treasure Island movies :')
ti = pd.DataFrame(ti, columns =['Title', 'Year'])
ti

List all of the Treasure Island movies :


Title  Year
0  Treasure Island  1918
1  Treasure Island  1920
2  Treasure Island  1934
3  Treasure Island  1950
4  Treasure Island  1972
5  Treasure Island  1973
6  Treasure Island  1985
7  Treasure Island  1999

# How many movies were made in the year 1950?

In [16]:
titles[titles['year'] == 1960]['title'].count()

1549

1549 movies were made in 1950.

# How many movies were made from 1950 through 1959?

In [17]:
titles[(titles['year'] >= 1950) & (titles['year'] <= 1959)]['title'].count()

12934

12934 movies were made between the years 1950 and 1959.

# In what years has a movie titled "Batman" been released?

In [18]:
bat = titles[titles['title'] == "Batman"]['year'].tolist()
bat.sort()
print("Batman has been released in the years", *bat)

Batman has been released in the years 1943 1989


# How many roles were there in the movie "Inception"?

In [21]:
cast=pd.read_csv("cast.csv")

In [22]:
roles=cast[cast['title'] == 'Inception']['type'].count()
print("The number of roles in inception are ",roles)

The number of roles in inception are  78


# How many roles in the movie "Inception" are NOT ranked by an "n" value?

In [27]:
non=cast[cast['title'] == 'Inception']['n'].isnull().sum()
print("The number of roles in the movie Inception that  are NOT ranked by an n value are ",non)

The number of roles in the movie Inception that  are NOT ranked by an n value are  27


# But how many roles in the movie "Inception" did receive an "n" value?

In [26]:
n=cast[cast['title'] == 'Inception']['n'].notnull().sum()
print("The number of roles in the movie Inception that  recieved an n value are ",n)

The number of roles in the movie Inception that  recieved an n value are  51


# Display the cast of "North by Northwest" in their correct "n"-value order, ignoring roles that did not earn a numeric "n" value.

In [28]:
cast1= cast[cast.n.notnull()].sort_values(by=['n'])
l = cast1[cast1['title'] == 'North by Northwest'][['name','n']].values.tolist()
print("The cast of 'North by Northwest' in their correct 'n'-value order,  ignoring roles that did not earn a numeric 'n' value:")
nbn = pd.DataFrame(l, columns = ['Name', 'n']) 
nbn

The cast of 'North by Northwest' in their correct 'n'-value order,  ignoring roles that did not earn a numeric 'n' value:


Name     n
0             Cary Grant   1.0
1        Eva Marie Saint   2.0
2            James Mason   3.0
3    Jessie Royce Landis   4.0
4         Leo G. Carroll   5.0
5   Josephine Hutchinson   6.0
6            Philip Ober   7.0
7          Martin Landau   8.0
8          Adam Williams   9.0
9           Edward Platt  10.0
10     Robert Ellenstein  11.0
11          Les Tremayne  12.0
12       Philip Coolidge  13.0
13         Patrick McVey  14.0
14          Edward Binns  15.0
15             Ken Lynch  16.0

# Display the entire cast, in "n"-order, of the 1972 film "Sleuth".

In [29]:
cast2 = cast[cast.n.notnull()].sort_values(by=['n'])
print('The entire cast, in "n"-order, of the 1972 film "Sleuth":')
cast2 = cast2.loc[(cast2['title']=='Sleuth') & (cast2['year']==1972)][['name','n']].reset_index(drop=True)
cast2

The entire cast, in "n"-order, of the 1972 film "Sleuth":


name    n
0    Laurence Olivier  1.0
1       Michael Caine  2.0
2      Alec Cawthorne  3.0
3  John (II) Matthews  4.0
4  Eve (III) Channing  5.0
5        Teddy Martin  6.0

# Now display the entire cast, in "n"-order, of the 2007 version of "Sleuth".

In [30]:
cast2 = cast[cast.n.notnull()].sort_values(by=['n'])
print('The entire cast, in "n"-order, of the 2007 film "Sleuth":')
cast2 = cast2.loc[(cast2['title']=='Sleuth') & (cast2['year']==2007)][['name','n']].reset_index(drop=True)
cast2

The entire cast, in "n"-order, of the 2007 film "Sleuth":


name    n
0  Michael Caine  1.0
1       Jude Law  2.0
2  Harold Pinter  3.0

# How many roles were credited in the silent 1921 version of Hamlet?

In [32]:
cast4 = cast.loc[(cast['title']=='Hamlet') & (cast['year']==1921)]
a=cast4['character'].count()
print("There were ",a,"Roles in silent 1921 version of Hamlet ")

There were  9 Roles in silent 1921 version of Hamlet 


# How many roles were credited in Branagh’s 1996 Hamlet?

In [33]:
cast4 = cast.loc[(cast['title']=='Hamlet') & (cast['year']==1996)]
b=cast4['character'].count()
print("There were ",b,"Roles in Branagh's 1996 version of Hamlet ")

There were  54 Roles in Branagh's 1996 version of Hamlet 


# How many "Hamlet" roles have been listed in all film credits through history?

In [34]:
cast5 = cast[cast['title'] == 'Hamlet']
c=cast5['character'].count()
print("There were ",c,"Roles in all  version of Hamlet ")

There were  342 Roles in all  version of Hamlet 


How many people have played an "Ophelia"?

In [35]:
print("The number of people that played role of  Ophelia are ",
cast[cast['character'] == 'Ophelia']['name'].count())

The number of people that played role of  Ophelia are  117


# How many people have played a role called "The Dude"?

In [37]:
print("The number of people that played role of  The Dude are ",cast[cast['character'] == 'The Dude']['name'].count())

The number of people that played role of  The Dude are  19


# How many people have played a role called "The Stranger"?

In [39]:
print("The number of people that played role of  The Stranger are ",cast[cast['character'] == 'The Stranger']['name'].count())

The number of people that played role of  The Stranger are  212


# How many roles has Sidney Poitier played throughout his career?

In [40]:
print("Sidney Poitier has played",cast[cast['name'] == "Sidney Poitier"]['character'].count(),"Roles throughout his career")

Sidney Poitier has played 43 Roles throughout his career


# How many roles has Judi Dench played?

In [41]:
print("Judi Dench has played",cast[cast['name'] == "Judi Dench"]['character'].count(),"Roles throughout his career")

Judi Dench has played 55 Roles throughout his career


# List the supporting roles (having n=2) played by Cary Grant in the 1940s, in order by year.

In [42]:
cast6 = cast[(cast['name'] == "Cary Grant") & (cast['n'] == 2) & (cast['year'] >= 1940) & (cast['year'] < 1950)]
li = cast6[['character','year']].values.tolist()
cg = pd.DataFrame(li, columns = ['Supporting Role', 'Year'])
cg

Supporting Role  Year
0      Nick Arden  1940
1     Roger Adams  1941

# List the leading roles that Cary Grant played in the 1940s in order by year.

In [43]:
cast7 = cast[(cast['name'] == "Cary Grant") & (cast['n'] == 1) & (cast['year'] >= 1940) & (cast['year'] < 1950)].sort_values(by = ['year'])
lis= cast7[['character','year']].values.tolist()
cg_lead = pd.DataFrame(lis, columns = ['Leading Role', 'Year'])
cg_lead

Leading Role  Year
0                            Matt Howard  1940
1                           Walter Burns  1940
2                     C. K. Dexter Haven  1940
3                       Johnnie Aysgarth  1941
4                           Leopold Dilg  1942
5                  Patrick 'Pat' O'Toole  1942
6                          Capt. Cassidy  1943
7   Joe Adams -posing as Joe Bascopolous  1943
8                            Jerry Flynn  1944
9                      Mortimer Brewster  1944
10                            Ernie Mott  1944
11                           Cole Porter  1946
12                                Devlin  1946
13                           Dick Nugent  1947
14                                Dudley  1947
15                         Jim Blandings  1948
16                     Dr. Madison Brown  1948
17                 Captain Henri Rochard  1949

# How many roles were available for actors in the 1950s?

In [47]:
print(cast[(cast['year'] >= 1950) & (cast['year'] < 1960) & (cast['type'] == 'actor')]['character'].count()," roles were available for actors in the 1950s")

157738  roles were available for actors in the 1950s


# How many roles were available for actresses in the 1950s?

In [52]:
print(cast[(cast['year'] >= 1950) & (cast['year'] < 1960) & (cast['type'] == 'actress')]['character'].count()," roles were available for actress in the 1950s")

57733  roles were available for actress in the 1950s


# How many leading roles (n=1) were available from the beginning of film history through 1980?

In [53]:
print(cast[(cast['year'] >= 1894) & (cast['year'] <= 1980) & (cast['n'] == 1)]['character'].count(),"roles (n=1) were available from the beginning of film history through 1980")

65140 roles (n=1) were available from the beginning of film history through 1980


# How many non-leading roles were available through from the beginning of film history through 1980?

In [54]:
print(cast[(cast['year'] >= 1894) & (cast['year'] <= 1980) & (cast['n'] > 1)]['character'].count(),"non-leading roles were available through from the beginning of film history through 1980")

669320 non-leading roles were available through from the beginning of film history through 1980


# How many roles through 1980 were minor enough that they did not warrant a numeric "n" rank?

In [55]:
print(cast[(cast['year'] >= 1894) & (cast['year'] <= 1980)]['n'].isnull().sum(),"roles through 1980 were minor enough that they did not warrant a numeric n rank")

448347 roles through 1980 were minor enough that they did not warrant a numeric n rank
